layers/ipsec.go

```diff
@@ -14,10 +14,10 @@ import (
 // IPSecAH is the authentication header for IPv4/6 defined in
 // http://tools.ietf.org/html/rfc2402
 type IPSecAH struct {
-	// While the auth header can be used for both IPv4 and v6, its format is that of
-	// an IPv6 extension (NextHeader, PayloadLength, etc...), so we use ipv6ExtensionBase
-	// to build it.
-	ipv6ExtensionBase
```

Comment:
how about keeping ipv6ExtensionBase up here, but not using decodeIPvExtensionBase below?

I think that's slightly more backwards-compatible.